orm/entities/Submission/TallySheet/__init__.py

```diff
@@ -42,12 +43,24 @@ def set_latest_version(self, tallySheetVersion: TallySheetVersion):
             self.submission.set_latest_version(submissionVersion=tallySheetVersion.submissionVersion)
 
     def set_locked_version(self, tallySheetVersion: TallySheetVersion):
+        if self.submittedStamp.createdBy == get_user_name():
```

Comment:
This has to be checked only if it's data entry @callkalpa

Reply:
Added the check.